bigbone/src/main/kotlin/social/bigbone/api/method/PushNotificationMethods.kt

```diff
@@ -0,0 +1,126 @@
+package social.bigbone.api.method
+
+import kotlinx.serialization.SerialName
+import kotlinx.serialization.Serializable
+import social.bigbone.MastodonClient
+import social.bigbone.MastodonRequest
+import social.bigbone.Parameters
+import social.bigbone.api.entity.WebPushSubscription
+import social.bigbone.api.exception.BigBoneRequestException
+
+/**
+ * Allows access to API methods with endpoints having an "api/vX/push" prefix.
+ * @see <a href="https://docs.joinmastodon.org/methods/push/">Mastodon push notification API methods</a>
+ */
+class PushNotificationMethods(private val client: MastodonClient) {
+
+    private val pushEndpoint = "/api/v1/push/subscription"
+
+    /**
+     * Specify whether to receive push notifications from all, followed, follower, or none users.
+     */
+    @Serializable
+    enum class PushDataPolicy {
+        @SerialName("all")
+        ALL,
+
+        @SerialName("followed")
+        FOLLOWED,
+
+        @SerialName("follower")
+        FOLLOWER,
+
+        @SerialName("none")
+        NONE
+    }
```

Comment:
I think we should either move this to a separate file within `entity` or make it `private` and remove the `@Serializable` and `SerialName` annotations. From what I can see, we only use this to determine parameters to send to the `POST` and `PUT` endpoints, but never to deserialise values we got from the API.

Reply:
You are right about the unity of adding kotlinx annotations, since it is only used to limit the range values of a param. By the way, by making this private, it would not be accessible from outside, such as Rx module. Moving this within entity package? I do not know... it is strictly associated to one param passed to the methods. Then, I think leaving within the Methods class it is the best choice.

Reply:
> By the way, by making this private, it would not be accessible from outside, such as Rx module.

Oh, yeah, good point! 👍 

> Moving this within entity package? I do not know... it is strictly associated to one param passed to the methods. Then, I think leaving within the Methods class it is the best choice.

OK 👍